### PR TITLE
Add team creation command

### PR DESCRIPTION
**Pull request by jos-b**
This PR adds a `!createteam` command to create a team channel for the code jam.

image

It creates the jams category if it does not exist, adds the jammer and team leader roles and then adds overwrites to the channel so that team members can see it and team leaders can pin and add Webhooks in it.

### CODE REVIEW

File: bot/__main__.py [hunk 1]
```diff
@@ -78,6 +78,7 @@
 bot.load_extension("bot.cogs.utils")
 bot.load_extension("bot.cogs.wolfram")
 bot.load_extension("bot.cogs.free")
+bot.load_extension("bot.cogs.jams")
```
Comment (jchristgit) [inline bot/__main__.py]: `:sort` please

File: bot/cogs/jams.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+from bot.constants import Roles
+from bot.decorators import with_role
+
+log = logging.getLogger(__name__)
+
+
+class CodeJams:
+    """
+    A cog for managing the code-jam related parts of our server
```
Comment (jchristgit) [inline bot/cogs/jams.py]: cog is implied. prefer `Code jam related management tasks`

Comment (heavysaturn) [reply bot/cogs/jams.py]:
```suggestion
    Manages the code-jam related parts of our server
```

File: bot/__main__.py (hunk 1 above)
Comment (jos-b) [reply bot/__main__.py]: `:sorted

**Pull request merged**